allenact/utils/experiment_utils.py

```diff
@@ -705,6 +712,17 @@ def reset(self):
         for memory in self.stage_component_uuid_to_stream_memory.values():
             memory.clear()
 
+    # TODO: Replace Any with the correct type
+    def change_engine_attributes(self, engine: Any):
+        if self.callback_to_change_engine_attributes is not None:
+            for key, value in self.callback_to_change_engine_attributes.items():
+                # check if the engine has the attribute
+                assert hasattr(engine, key)
+
+                func = value["func"]
+                args = value["args"]
+                setattr(engine, key, func(engine, **args))
```

Comment:
LGTM. I assume func is also responsible to ensure consistency of the engine state?

Reply:
yes, Luca also mentioned that people might mess the engine by the `func`.
But I decided to have user to be responsible to ensure consistency for now.